ethereum.star

```
@@ -9,8 +10,31 @@ GETH_IMAGE = "ethereum/client-go:v1.15.11"
 # https://github.com/ethpandaops/ethereum-package/pull/915
 LIGHTHOUSE_IMAGE = "ethpandaops/lighthouse:stable-999b045"
 
+only_smc_genesis = "templates/genesis/only-smc-deployed-genesis.json"
+op_rollup_created_genesis = "templates/genesis/op-genesis.json"
+
 
 def run(plan, args):
+    if args.get("custom_genesis") == True:
+        if args.get("consensus_contract_type") == constants.CONSENSUS_TYPE.pessimistic:
+            plan.print(
+                "Custom genesis is enabled with pessimistic consensus, using the forked ethereum package for pessimistic."
+            )
+            genesis = read_file(src=op_rollup_created_genesis)
+        elif (
+            args.get("consensus_contract_type") == constants.CONSENSUS_TYPE.cdk_validium
+            or args.get("consensus_contract_type") == constants.CONSENSUS_TYPE.rollup
+        ):
+            plan.print(
+                "Custom genesis is enabled for rollup/validium consensus, using the forked ethereum package without any rollup deployed."
+            )
+            genesis = read_file(src=only_smc_genesis)
+        else:
+            plan.print("Unknown consensus contract type")
+            return
```

maybe we should trigger an error here with something like `fail("Unknown consensus contract type")` instead of an empty `return`

---

Sure, I didn't know it exists xD